neurons/validator.py

```diff
@@ -588,8 +586,8 @@ async def run_step(self):
                 hotkey
             )
 
-            old_losses = [math.inf for _ in range(len(old_batches))]
```

Comment:
The batches will be different lengths since you load the same number of pages but use different sequence lengths.

As a higher level comment, I'd recommend first prohibiting models prior to the 7B block (reject them in the update loop) and then you can remove a lot of the complexity in this function since you won't have to handle the new and old tokenizers. If that seems too expensive to do now, then perhaps something to think about for the next release

Reply:
I agree. In the original code, Release v3.1.4, there was no distinction between new and old losses. We introduced this distinction in 3.2.0 (the version with the bug), then I think that we should remove the distinction again. The way I understand it is that, if the model does not have metadata (line592) then we are no longer able to distinguish whether this is an old or new model, so we use whatever loss list (regardless, new or old), because it will be set to infinity.

If the model has meta data, the `losses` variable is completely overwritten in line 627.

Reply:
I agree, we discussed removed support to models prior to 7B, but for the next release

Reply:
In the exceptional case line 627 does not get hit and you instead hit 641 which uses the losses as they are. Then in 655 when you compute wins you could have mismatched lengths of the losses.